src/icrar/leap-accelerate/math/linear_math_helper.h

```diff
@@ -62,6 +64,7 @@ namespace icrar
 
         auto m = Eigen::Matrix<T, R, C>();
 
+        // column major to column major
```

Comment:
If these copying operations of data are all column-major to column-major you could simply use `std::copy()` instead of the hand-rolled loops. Or is the underlying storage differently organize, and it's column- to row-major?

Reply:
I worked out both are column major after playing around with them for a while. Some matrixes from Eigen can be made row major using a template flag and they get marked different when printed out.

I'll give std::copy a go as tests should break now if the behaviour changes

Reply:
I think the iterator concept gets disabled with a static assert for 2D Eigen::Matrix
```
error: invalid use of void expression
   66 |         std::copy(value.begin(), value.end(), m.begin());
...
/usr/local/include/eigen3/Eigen/src/Core/StlIterators.h:407:3: error: static assertion failed: YOU_TRIED_CALLING_A_VECTOR_METHOD_ON_A_MATRIX
  407 |   EIGEN_STATIC_ASSERT_VECTOR_ONLY(Derived);
```

Another approach though is a c-style memcpy() is likely the fastest